### PR TITLE
#159951983 Users should be able to report an article

### PR DESCRIPTION
**Pull request by byarustev**
#### What does this PR do?
implements functionality for reporting an article 
#### Description of Task to be completed?
- reporting an article POST `/api/article/<slug>/report`
- viewing reports to a given article  GET `/api/article/<slug>/report`
#### How should this be manually tested?
- to report an article in postman, send a POST request replacing <slug> with the slug of the article.
- add a report message in the body 
```
 {
   "message": " this is the reason why I am reporting this article"
 }
```
- to view reports to an article send the send a GET request to the same url but using a token for a super admin

#### What are the relevant pivotal tracker stories?
#159951983
#### Screenshot
screen shot 2018-09-18 at 15 43 39

### CODE REVIEW

File: authors/apps/articles/serializers.py [hunk 1]
```diff
@@ -243,3 +226,19 @@ def handle_existence(instance, data):
...
+        model = ReportedArticles
+        fields = ['user', 'article', 'message']
+
+    def create(self, validated_data):
+        return ReportedArticles.objects.create(**validated_data)
+
+    def to_representation(self, instance):
+        response = super().to_representation(instance)
+        response["user"] = instance.user.username
+        response["article"] = response["article"] = instance.article.slug
```
Comment (cmplx-xyttmt) [inline authors/apps/articles/serializers.py]: @byarustev , the second `response["article"]` here is not necessary.

Comment (byarustev) [reply authors/apps/articles/serializers.py]: sorted

**Comment by cmplx-xyttmt** [comment]
LGTM

**Comment by ruganda** [comment]
LGTM

**Comment by marthamareal** [comment]
@byarustev  you realize that in migrations you are deleting a model for FavoriteArticles

**Comment by byarustev** [comment]
> @byarustev you realize that in migrations you are deleting a model for FavoriteArticles
not really all models are in the migrations file

**Comment by marthamareal** [comment]
> > @byarustev you realize that in migrations you are deleting a model for FavoriteArticles
> > not really all models are in the migrations file

check in your changed files. we have to add on migrations not replace.

**Comment by marthamareal** [comment]
LGTM

**Pull request merged**